### PR TITLE
fix failing to grab local IP

### PR DESCRIPTION
**Pull request by yoachim**
Make it so it doesn't fail if I try to run/install at home.

### CODE REVIEW

File: tests/testEndToEnd.py [hunk 1]
```diff
@@ -135,14 +135,20 @@ def get_mag(self):
...
-        else:
-        #works on linux, at least OpenSuSE
-            s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-            s.connect(('8.8.8.8', 0))  # connecting to a UDP address doesn't send packets
-            local_ip_address = s.getsockname()[0]
+        # Make it pass on a plane
+        try:
+            if sys.platform == 'darwin':
+                s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+                s.connect(("lsst.org", 80))
```
Comment (danielsf) [inline tests/testEndToEnd.py]: for my own edification: what does this 'lsst.org' refer to?

**Comment by yoachim** [comment]
It failed when I tried to run it at home because then the machine name was something like "Peters-macbook.local", and thus couldn't be resolved to an IP address. The solution I found on StackOverflow was to (I think) connect to a web server and then see what IP address that web server sent a reply to. Usually people put in `google.com`, but it's considered rude to hit a 3rd party server for your tests, and since we have our own project web page I figured that would be an obvious substitute.

**Comment by danielsf** [comment]
Doesn't this mean that anyone trying to install while offline (i.e. if I cannot ping lsst.org), will experience a failure?

**Comment by danielsf** [comment]
I will test that later and open a ticket to fix it, if necessary.

**Comment by yoachim** [comment]
I wrapped it in a try/except, so it should skip the test if offline, but I don't think I tested it, so it might fail in some other way.

**Comment by danielsf** [comment]
Right.  I will test it tonight.

**Pull request merged**